xarray/core/dataset.py

```diff
@@ -4053,13 +4053,68 @@ def drop_sel(self, labels=None, *, errors="raise", **labels_kwargs):
                 labels_for_dim = [labels_for_dim]
             labels_for_dim = np.asarray(labels_for_dim)
             try:
-                index = self.indexes[dim]
+                index = self.get_index(dim)
```

Comment:
Out of interest, what's the difference between these?

Reply:
The idea is that the new code will work in dimensions that have no coordinates

Reply:
I might be missing something, but I don't think `drop_sel` should work for dimensions without coordinates: `sel` as well as `drop_sel` work on labels, so they need the coordinates. For reference, `self.get_index(dim)` returns a `DataArray` containing the index positions, so this actually makes `drop_sel` behave like `drop_isel` for dimensions without coordinates.

Reply:
`xr.DataArray(np.arange(10)).sel(dim_0=[2, 3])` This works though so it makes sense that `drop_sel` works in all the cases that `sel` does.